# A/B Testing 
It is a very common technique to evaluate effectiveness of changes in a user interface. You show some users interface A (an existing design), and to others you show interface B (a newer design), and then check if people search more with the new design or still prefer the older design. The attached dataset “searches.json” contains data from one such A/B testing experiment. 
Each line in this file represents data of one user’s interactions. The following data is collected for each user: 
● uid: unique user id - users with an odd-numbered uid were shown the new design (B) 
● is_instructor: Instructors are testers or presenters who demonstrate search functionality 
to prospective clients. 
● login_count: number of times the user logged in 
● search_count: number of times the user used the search feature 
Write a program that answers the questions: 
● Did more users use the search feature in the new design (B)? 
● Did users search more often in the new design (B)? 
 

# Total number of searches
### Comparison between new and old design

In [15]:
import json
import pandas as pd
import numpy as np

location = 'searches.json' #location to import data
df = pd.read_json(location, lines= True)

# creating a new column by 'is_new_desing' name
df['is_new_design'] = np.where((df['uid']%2 != 0), 'yes','no' )
df.head()

is_instructor  login_count  search_count       uid is_new_design
0           True            1             2   6061521           yes
1          False            4             0  11986457           yes
2          False            1             0  15995765           yes
3           True            3             0   9106912            no
4          False            1             0   9882383           yes

In [16]:
# grouping necessary information
df.groupby(['is_new_design', 'is_instructor'])['search_count'].sum()

is_new_design  is_instructor
no             False            145
               True             121
yes            False            143
               True              88
Name: search_count, dtype: int64

In [6]:
# no. of searches 
new_dsn_tot_src_count= df[(df['is_instructor'] == False)& (df['is_new_design'] == 'yes')].search_count.sum()
print('total number of searches with new design interface is', new_dsn_tot_src_count)

total number of searches with new design interface is 143


In [7]:
#no. of searches
old_dsn_tot_src_count= df[(df['is_instructor'] == False)& (df['is_new_design'] == 'no')].search_count.sum()
print('total number of searches with old design interface is', old_dsn_tot_src_count)

total number of searches with old design interface is 145


In [9]:
#creating new dataframes
#removing rows with zero search count the rows with instructur searches
df_new_dsn_src_used = df[(df['is_instructor'] == False)& (df['is_new_design'] == 'yes')&(df['search_count']>0)]
df_old_dsn_src_used = df[(df['is_instructor'] == False)& (df['is_new_design'] == 'no')&(df['search_count']>0)]

In [18]:
new_dsn_src_used = len(df_new_dsn_src_used)
print('Number of users used search feature in new design(B) is ', new_dsn_src_used)

Number of users used search feature in new design(B) is  65


In [17]:
old_dsn_src_used = len(df_old_dsn_src_used)
print('Number of users used search feature in old design(B) is ', old_dsn_src_used)

Number of users used search feature in old design(B) is  61


# Conclusion


## More number of users(65) used search features in new design(B)(65) compared to  old design(A) (61)

## Old design interface has more number of total searches(145) compared to new design total searches (143)